server/routes/bookings/cancelVisitController.ts

```diff
@@ -0,0 +1,98 @@
+import type { RequestHandler } from 'express'
+import { Meta, ValidationChain, matchedData, param, body } from 'express-validator'
+import { BookerService, VisitService } from '../../services'
+import paths from '../../constants/paths'
+
+export default class CancelVisitController {
+  public constructor(
+    private readonly bookerService: BookerService,
+    private readonly visitService: VisitService,
+  ) {}
+
+  public confirmCancelView(): RequestHandler {
+    return async (req, res) => {
+      const { booker, bookings } = req.session
+      const { visits } = bookings
+
+      const prisoner = booker.prisoners?.[0]
+        ? booker.prisoners?.[0]
+        : (await this.bookerService.getPrisoners(booker.reference))?.[0]
+
+      const { visitDisplayId } = matchedData<{ visitDisplayId: string }>(req)
+
+      const visit = visits.find(v => v.visitDisplayId === visitDisplayId)
+
+      const nowTimestamp = new Date()
```

Comment:
Maybe a util fn as you're doing this twice? Just pass the comparison date and return true/false?